backend/prisma/schema.prisma

```diff
@@ -40,20 +45,49 @@ model GroupMember {
   addedById Int
   addedAt   DateTime  @default(now())
   group     CodeGroup @relation("GroupMembers", fields: [groupId], references: [id])
-  user      User      @relation("UserGroupMemberships", fields: [userId], references: [id])
-  addedBy   User      @relation("AddedByUser", fields: [addedById], references: [id])
+  user      User      @relation("GroupsAddedTo", fields: [userId], references: [id])
 
   @@unique([groupId, userId])
 }
 
 model File {
   id            Int       @id @default(autoincrement())
   fileName      String
-  fileContent   String
+  fileContent   String?
   creatorId     Int
   groupId       Int
   createdAt     DateTime  @default(now())
   lastUpdatedAt DateTime  @updatedAt
-  user          User      @relation("UserFiles", fields: [creatorId], references: [id])
+  creator       User      @relation("UserFiles", fields: [creatorId], references: [id])
   codeGroup     CodeGroup @relation("GroupFiles", fields: [groupId], references: [id])
 }
+
+model Skill {
+  id          Int        @id @default(autoincrement())
+  skill       String
+  user        User?      @relation(fields: [userId], references: [id])
+  userId      Int?
+  CodeGroup   CodeGroup? @relation(fields: [codeGroupId], references: [id])
+  codeGroupId Int?
```

Comment:
Why are skills linked to specific code groups? Or is the setup that a skill can be for a user or for a group?

Reply:
For the recommendation algorithm, I will be comparing the user's skills to the preferred_skillls of that specific codegroup. So I thought it was important that I link them this way.